src/python/pants/backend/jvm/tasks/jvm_dependency_analyzer.py

```diff
@@ -83,8 +83,9 @@ def targets_for_class(self, target, classname):
     """Search which targets from `target`'s transitive dependencies contain `classname`."""
     targets_with_class = set()
     for target in target.closure():
-      if classname in self._target_classes(target):
-        targets_with_class.add(target)
+      for one_class in self._target_classes(target):
+        if classname in one_class:
```

Comment:
Do you have a sense for how much more expensive this is relative to doing the set lookups? For targets with large dependency graphs, adding this inner loop might make missing-deps-suggest a fair bit slower.

If it is slower, it might make sense to split doing the partial matching into its own review to focus on changing that.

Reply:
I don't think a separate review is necessary. But I agree that benchmarking would be a good idea.

Reply:
Also, this should have a break in it, when it matches

Reply:
I benchmarked these with a few small to large targets and found that there is a ~2 second duration increase to complete missing-deps-suggest across all of the targets. It seems that the slow down is related to parsing rather than the additional looping.